config/config.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	DSN      string
+	InfoLog  *log.Logger
+	ErrorLog *log.Logger
+	DB       *sql.DB
+	Addr     *string
+}
+
+// Init the application's configuration
+func Init(infoLog, errorLog *log.Logger) (*Config, error) {
+
+	//Read config file
+	viper.SetConfigName("config")
+	viper.SetConfigType("json")
+	viper.AddConfigPath(".")
+	err := viper.ReadInConfig()
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found
+			panic(fmt.Errorf("fatal error config file not found: %s", err))
```

Comment:
Better to handle all panics at the top level instead of here.

Your function returns an error; use that at your favor.